pkg/sources/filesystem/filesystem.go

```diff
@@ -179,13 +307,31 @@ func (s *Source) scanDir(ctx context.Context, path string, chunksChan chan *sour
 
 var skipSymlinkErr = errors.New("skipping symlink")
 
+// isDirectChild checks if a path is a direct child of any scan root path.
+// This enforces depth-1 symlink following to prevent:
+// - Infinite symlink loops
+// - Deep directory traversal through symlinks
+//
+// Returns true only if the symlink's parent directory matches a scan root path.
+func (s *Source) isDirectChild(path string) bool {
+	dir := filepath.Clean(filepath.Dir(path))
+	_, isRoot := s.scanRootPaths[dir]
+	return isRoot
+}
```

Comment:
When I read "depth-1 limiting" I thought it meant we allow only one symlink depth, but this looks like it is limiting symbolic link targets to a very specific directory structure.

Why are we doing this? It seems like something that can be easily tripped on and I'm not seeing why it needs to be that way.